src/wp-includes/compat.php

```diff
@@ -478,6 +478,33 @@ function str_ends_with( $haystack, $needle ) {
 	}
 }
 
+if ( ! function_exists( 'array_is_list' ) ) {
+	/**
+	 * Polyfill for `array_is_list()` function added in PHP 8.1.
+	 *
+	 * Determines if the given array is a list.
+	 *
+	 * An array is considered a list if its keys consist
+	 * of consecutive numbers from 0 to count( $array ) - 1.
+	 *
+	 * This function returns `true` on empty arrays.
+	 *
+	 * @since 6.0.0
+	 *
+	 * @param array $arr The array.
+	 * @return bool True if array is a list, false otherwise.
+	 */
+	function array_is_list( array $arr ) {
+		$i = 0;
+		foreach ( $arr as $k => $v ) {
+			if ( $k !== $i++ ) {
```

Comment:
@costdev sorry if I am wrong, I think on line 500 it will be if( $k !== $i ) then return false.

Reply:
Hi @rafiahmedd! It is actually a short version of doing this:

```php
if ( $k !== $i ) {
    return false;
}
$i++
```

- `$i++` means `return $i, then increment $i by one`
- `++$i` means `increment $i by one, then return $i`

See [the PHP manual entry](https://www.php.net/manual/en/language.operators.increment.php) on incrementation for more information, and you can [see it in action here](https://3v4l.org/LBEKW).